### PR TITLE
[tests/platform/test_reboot.py] add testcases for reboot cause

### PR DESCRIPTION
**Pull request by stephenxs**
### Description of PR

Summary:
Fixes # (issue)
implement [reboot-cause testcases](https://github.com/stephenxs/SONiC/pull/2)

### Type of change

- [] Bug fix
- [] Testbed and Framework(new/improvement)
- [x] Test case(new)

#### How did you do it?
add hardware reboot testcases to reboot test
add reboot-cause checking after each reboot.

#### How did you verify/test it?
run test on every testbed

### CODE REVIEW

File: tests/platform/test_reboot.py [hunk 1]
```diff
@@ -37,19 +53,33 @@ def reboot_and_check(localhost, dut, interfaces, reboot_type="cold"):
     elif reboot_type == "warm":
         reboot_cmd = "warm-reboot"
         reboot_timeout = 180
+    elif reboot_type == "power off" or reboot_type == "watchdog":
```
Comment (wangxin) [inline tests/platform/test_reboot.py]: If just issue a command can perform "watchdog" reboot, then the watchdog rebooting and checking can be just like reboot_types "cold", "warm" or "fast". For example:
```
    elif reboot_type == "watchdog":
        reboot_cmd = "python -c \"import sonic_platform.platform as P; P.Platform().get_chassis().get_watchdog().arm(5); exit()\""
        reboot_timeout = 180
```

File: tests/platform/test_reboot.py [hunk 2]
```diff
@@ -21,13 +21,29 @@
...
+        reboot_cause_got = m.group(1)
+    assert reboot_cause_expected == reboot_cause_got, \
+        "got reboot-cause %s after rebooted by %s" % (reboot_cause_got, reboot_cause_expected)
+
+def reboot_and_check(localhost, dut, interfaces, reboot_type="cold", reboot_helper=None, reboot_argu=None):
     """
     Perform the specified type of reboot and check platform status.
     """
     logging.info("Run %s reboot on DUT" % reboot_type)
+    hardware_reboot = False
```
Comment (wangxin) [inline tests/platform/test_reboot.py]: The hardware_reboot variable is not really necessary

File: tests/platform/test_reboot.py [hunk 3]
```diff
@@ -117,3 +152,59 @@ def test_warm_reboot(testbed_devices, conn_graph_facts):
             pytest.skip("ISSU is not supported on this DUT, skip this test case")
 
     reboot_and_check(localhost, ans_host, conn_graph_facts["device_conn"], reboot_type="warm")
+
+def _power_off_reboot_helper(argu):
```
Comment (wangxin) [inline tests/platform/test_reboot.py]: Use name "args" is better.

File: tests/platform/test_reboot.py [hunk 4]
```diff
@@ -117,3 +152,59 @@ def test_warm_reboot(testbed_devices, conn_graph_facts):
...
+    power_on_seq = argu["power_on_seq"]
+    delay_time = argu["delay_time"]
+
+    for psu in all_psu:
+        psu_ctrl.turn_off_psu(psu["psu_id"])
+    time.sleep(delay_time)
+    logging.info("Power on {}".format(power_on_seq))
+    for psu in power_on_seq:
+        psu_ctrl.turn_on_psu(psu["psu_id"])
+
```
Comment (wangxin) [inline tests/platform/test_reboot.py]: Two emply lines between functions according to PEP8 style guide.

File: tests/platform/test_reboot.py (hunk 1 above)
Comment (stephenxs) [reply tests/platform/test_reboot.py]: fixed.

File: tests/platform/test_reboot.py (hunk 2 above)
Comment (stephenxs) [reply tests/platform/test_reboot.py]: removed

**Comment by stephenxs** [comment]
community PR has been created [[tests/platform/test_reboot.py] add testcases for reboot cause #1079](https://github.com/Azure/sonic-mgmt/pull/1079)

**Pull request closed without merging**